src/pczt.rs

```diff
@@ -0,0 +1,533 @@
+    ///
+    /// Entries are added by the Constructor, and modified by an Updater, IO Finalizer,
+    /// Signer, Combiner, or Spend Finalizer.
+    pub(crate) actions: Vec<Action<D>>,
+
+    /// The flags for the Orchard bundle.
+    ///
+    /// This is set by the Creator. The Constructor MUST only add spends and outputs that
+    /// are consistent with these flags (i.e. are dummies as appropriate).
+    pub(crate) flags: Flags,
+
+    /// The sum of the values of all `actions`.
+    ///
+    /// This is initialized by the Creator, and updated by the Constructor as spends or
+    /// outputs are added to the PCZT. It enables per-spend and per-output values to be
+    /// redacted from the PCZT after they are no longer necessary.
+    pub(crate) value_sum: ValueSum,
+
+    /// Assets intended for burning
+    pub(crate) burn: Vec<(AssetBase, NoteValue)>,
+
+    /// The Orchard anchor for this transaction.
+    ///
+    /// Set by the Creator.
+    pub(crate) anchor: Anchor,
+
+    /// Block height after which this Bundle's Actions are invalid by consensus.
+    ///
+    /// For the OrchardZSA protocol, `expiry_height` is set to 0, indicating no expiry.
+    /// This field is reserved for future use.
+    pub(crate) expiry_height: u32,
+
+    /// The Orchard bundle proof.
+    ///
+    /// This is `None` until it is set by the Prover.
+    pub(crate) zkproof: Option<Proof>,
+
+    /// The Orchard binding signature signing key.
+    ///
+    /// - This is `None` until it is set by the IO Finalizer.
+    /// - The Transaction Extractor uses this to produce the binding signature.
+    pub(crate) bsk: Option<redpallas::SigningKey<Binding>>,
+}
+
+impl<D: OrchardDomainCommon> Bundle<D> {
+    /// Returns a mutable reference to the actions in this bundle.
+    ///
+    /// This is used by Signers to apply signatures with [`Action::sign`].
+    ///
+    /// Note: updating the `Action`s via the returned slice will not update other
+    /// fields of the bundle dependent on them, such as `value_sum` and `bsk`.
+    pub fn actions_mut(&mut self) -> &mut [Action<D>] {
+        &mut self.actions
+    }
+}
+
+/// PCZT fields that are specific to producing an Orchard action within a transaction.
+///
+/// This struct is for representing Orchard actions in a partially-created transaction.
+/// If you have a fully-created transaction, use [the regular `Action` struct].
+///
+/// [the regular `Action` struct]: crate::Action
+#[derive(Debug, Getters)]
+#[getset(get = "pub")]
+pub struct Action<D: OrchardDomainCommon> {
+    /// A commitment to the net value created or consumed by this action.
+    pub(crate) cv_net: ValueCommitment,
+
+    /// The spend half of this action.
+    pub(crate) spend: Spend,
+
+    /// The output half of this action.
+    pub(crate) output: Output<D>,
+
+    /// The asset id of this Action.
+    ///
+    /// - This is set by the Constructor.
+    /// - This is required by the Prover.
+    pub(crate) asset: Option<AssetBase>,
```

Comment:
why Option?

Reply:
Like other values such as recipient, value, and rho, I set it as optional. The goal is to allow their removal once they are no longer needed, specifically after the proof has been created and the sighash has been evaluated.